### ISSUE TITLE
DataScope variables: No publish possible

### ISSUE THREAD

**Issue by NKarolak** [microsoft/AL#6620]
**1. Describe the bug**
It is not possible to publish if an object contains a global or protected variable of [data type DataScope](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/datascope/datascope-option).
Compilation / package build itself is no problem.
Error on publish:
```
Failure while emitting object. Object:'Codeunit "Your Object Name"'
This program location is thought to be unreachable. Normal variable of type DataScope
```

**2. To Reproduce**
1. Add either
```
var
  MyDataScope: DataScope;
```

or

```
protected var
  MyDataScope: DataScope;
```
To any codeunit of your choice - bot not to procedures!

2. Publish.

**3. Expected behavior**
Publish should be possible. If a good reason exists to not let it publish, please fix the compiler to prevent declaring such variables.

**4. Actual behavior**
Publish fails.
Note: Other Option Type variables do not fail on publish, if declared globally. Tested with `MyAction: Action;` and `MyClientType: ClientType;`

**5. Versions:**

- AL Language: 7.1.453917 / runtime 6.0
- Visual Studio Code: 1.56.0
- Business Central: 17.2

**Comment by NKarolak** [comment]
@kalberes 
Happy New Year!
This issue seems to be of low priority for you. However, do you have some ETA?

**Comment by kalberes** [comment]
Yes it is a low priority. Sorry for that. No ETA. If this is extra important to be fixed than the best way is to use the official channels that we get a Request For Hotfix.

**Comment by SBalslev** [comment]
Hi! Thank you for reporting this issue with DataScope variables preventing publish.

We've confirmed this was fixed in **BC 14.0** — a compiler crash when using DataScope as a local variable or return value was resolved. This should no longer be an issue on any currently supported version.

If you encounter similar problems on a current version, please don't hesitate to open a new issue.

Thank you for helping us improve the AL developer experience!
